app/cloud.go

```diff
@@ -18,6 +21,75 @@ func (a *App) getSysAdminsEmailRecipients() ([]*model.User, *model.AppError) {
 	return a.GetUsers(userOptions)
 }
 
+// SendAdminsOverTheLimitAlertEmail takes the username of user trying to alert admins and then applies rate limit of n (number of admins) emails per user per day
+// before sending the emails.
+func (a *App) SendAdminsOverTheLimitAlertEmail(username string, subscription *model.Subscription) *model.AppError {
```

Comment:
Also, given that we do actually send an email to the admin if they go over the limit, could we rename here to maybe `sendAdminUpgradeRequestEmail`? This would cover the case for `invite` and `joining` as well